src/middleware/basicAuthentication.coffee

```diff
@@ -51,6 +51,7 @@ exports.authenticateUser = (ctx, done) ->
         done null, null
   else
     logger.error "No basic auth details supplied"
+    ctx.authenticated = {} # Set to empty object rather than null
```

Comment:
Yeah, but I think you should change the middleware then. Basically the middleware should always call yield next but only sometimes is a authenticated user set. As it is now, it will never get into the else statement of the middleware and that else block can probably be removed.